src/dict.c

```diff
@@ -675,23 +694,36 @@ void dictRelease(dict *d)
 
 dictEntry *dictFind(dict *d, const void *key)
 {
-    dictEntry *he;
-    uint64_t h, idx, table;
+    dictEntry *he = NULL;
+    uint64_t h, idx;
 
     if (dictSize(d) == 0) return NULL; /* dict is empty */
-    if (dictIsRehashing(d)) _dictRehashStep(d);
+
     h = dictHashKey(d, key);
-    for (table = 0; table <= 1; table++) {
+    idx = h & DICTHT_SIZE_MASK(d->ht_size_exp[0]);
+    if ((long)idx >= d->rehashidx) {
+        he = d->ht_table[0][idx];
+    }
+
+    if (dictIsRehashing(d)) {
+        if ((long)idx >= d->rehashidx && he) {
+            dictBucketRehash(d, h);
+        /* If the 'he' is not in ht0, we perform a random bucket rehash. */
```

Comment:
i suppose this meant to document the `else` block, so let's move it inside it, and indent it further.
p.s. i'd like a comment in both branches of the if-else